src/Bridge/Doctrine/Orm/Util/IdentifierManagerTrait.php

```diff
@@ -39,6 +39,8 @@
      */
     private function normalizeIdentifiers($id, ObjectManager $manager, string $resourceClass): array
     {
+        @trigger_error(sprintf('Trait "%s" will be removed in version 3.0. Using it is deprecated since 2.2.', IdentifierManagerTrait::class), E_USER_DEPRECATED);
```

Comment:
This trait is internal, so there is no need to deprecate it, you can even remove it if you want to.

Reply:
If I remove the trait, injecting the IdentifierNormalizer will be mandatory

Reply:
All the dependencies of the composite identifier normalizer are already injected in `SubresourceDataProvider`, you can just initialize the class in the constructor if `null` is passed (and trigger a deprecation notice). The other benefit of doing this is that we'll have only one implementation to maintain.

Reply:
What about this one @soyuka?